CONTRIBUTING.md

```diff
@@ -1,48 +1,68 @@
+happen through Pull Requests submitted through Git.
 
 
 ## Requirements
 
-To contribute to ECS the following tools are expected to be running on your machine:
+You need these tools to contribute to ECS:
 
 * [Git](https://git-scm.com/)
 * [Python](https://www.python.org/)
 * [Go](https://golang.org/)
 
 ## Steps to contribute
 
-To contribute changes to ECS follow the steps below:
-
-* Create [your own fork](https://help.github.com/articles/fork-a-repo/) of the ECS repo.
-* Clone your fork to your machine.
-* Run `git checkout -b branch-name` and replace `branch-name` according to your change
-* Apply your changes to the `.yml` files as needed
-* Run `make` to update generated files like `schema.csv` and `schema.md`
-* Run `git commit -a -m "your message"`
-* Run `git push --set-upstream origin branch-name` (assumes your clone remote is called `origin`)
-* Create a [Pull Request](https://help.github.com/articles/creating-a-pull-request/) against the ECS repo. (if you go to the website, it should pop up as a button with your branch directly)
-* Wait for reviews on your PR and collaborate.
-
-Notes: Make sure to always only push changes against your own fork.
-
+Here are the steps for contributing to ECS.
+
+* Set up your git environment.
+  - Create [your own fork](https://help.github.com/articles/fork-a-repo/) of the ECS repo.
+  - Clone your fork to your machine.
+* Create a local branch to hold your changes.
+  - Run `git checkout -b branch-name`, where `branch-name` is the name you want to give your local branch
+* Do your work. 
+  - Make changes to the `.yml` files as needed
+* Run `make` to update generated files. 
+  - Note that the README.md file is generated, and should not be edited directly. Source files are in the /docs directory.
+* Commit your changes locally. 
+  - Run `git commit -a -m "your message"`
+* Push your changes to your own github.com fork.
+  - Run `git push --set-upstream origin branch-name` 
+  - In this command, `origin` is an alias that references your fork.
+* Request feedback and permission to merge your changes. 
+  - Create a [Pull Request](https://help.github.com/articles/creating-a-pull-request/) against the ECS repo. 
+  - (Look for the `Compare & pull request` button on your branch in github.com.)
+* Next steps
+  - Wait for reviews on your PR.
+  - Incorporate review comments and push updates if needed.
+* Thank you for your contribution!
+  
+**Important:** Be sure to push changes only to your own fork. Changes must be approved before they are merged into the main repository.
+
+### Other ways to contribute
+
+You can contribute even if you are not an experienced Git user. You'll need a github.com account.
+* Go to the ECS repo: https://github.com/elastic/ecs
+* Click `Issues` in the nav bar under the repo name.
+* Click `New issue`.
+* Add your text, and click `Submit new issue`.
```

Comment:
Let's add a note here that users should provide as many details as possible in the description of an issue. Sounds obvious but ...